compute/managed-instances/demo/main.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python3
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+from os import path
+from requests import get
+from re import sub
+from socket import gethostname
+from socketserver import ThreadingMixIn
+
+PORT_NUMBER = 80
+HTML_TEMPLATES_PATH = path.dirname(path.realpath(__file__)) + '/templates'
+HEALTHY = True
+
+env = Environment(
+    loader=FileSystemLoader(HTML_TEMPLATES_PATH),
+    autoescape=select_autoescape('html')
+)
+
+
+class request_handler(BaseHTTPRequestHandler):
+    def get_zone(self):
+        r = get('http://metadata.google.internal/'
+                'computeMetadata/v1/instance/zone',
+                headers={'Metadata-Flavor': 'Google'})
+        if r.status_code == 200:
+            return sub(r'.+zones/(.+)', r'\1', r.text)
+        else:
+            return ''
+
+    def get_template(self):
+        r = get('http://metadata.google.internal/'
+                'computeMetadata/v1/instance/attributes/instance-template',
+                headers={'Metadata-Flavor': 'Google'})
+        if r.status_code == 200:
+            return sub(r'.+instanceTemplates/(.+)', r'\1', r.text)
+        else:
+            return ''
+
+    def do_GET(self):
+        global HEALTHY
+
+        if self.path == '/makeHealthy':
+            HEALTHY = True
+            self.send_response(302)
+            self.send_header('Location', '/')
+            self.end_headers()
+
+        elif self.path == '/makeUnhealthy':
+            HEALTHY = False
+            self.send_response(302)
+            self.send_header('Location', '/')
+            self.end_headers()
+
+        elif self.path == '/health':
+            if not HEALTHY:
+                self.send_response(500)
+                self.end_headers()
+            else:
+                self.send_response(200)
+                self.send_header('Content-type', 'text/html')
+                self.end_headers()
+            html = env.get_template('health.jinja2').render(healthy=HEALTHY)
+            self.wfile.write(html.encode('UTF-8'))
+
+        else:
+            HOSTNAME = gethostname()
+            ZONE = self.get_zone()
+            TEMPLATE = self.get_template()
+
+            self.send_response(200 if HEALTHY else 500)
```

Comment:
In this case, is it intended behavior that we may be returning a 500 response but with the exact same template and information as if it were 200?

Reply:
I believe this was a coding mistake.  This represented the root path '/'.  I've fixed this in the Flask refactoring.